templates/test/ci/prow-external-cloud-provider/kustomization.yaml

```diff
@@ -2,10 +2,11 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 namespace: default
 resources:
-  - ../../../flavors/external-cloud-provider
+  - ../../../flavors/external-cloud-provider-machinepool
```

Comment:
would it make sense to just have one template that has both md and mp to test OOT cloud provider? Seems like we're getting rid of md by doing this? The base prow template has both I believe

Reply:
Also, are we testing both Flex and Uniform?

Reply:
I reverted this changed and instead created a new `prow-external-cloud-provider-machinepool` test template. It inherits from the `external-cloud-provider-machinepool` flavor, so they're both just testing VMSS Flex with Linux currently.

I also added a new e2e test spec for this. I would like to expand it to include both Flex and Uniform for Linux and Windows (4 node pools total), but it seems safer to start small.